Faster blocks

mononaut opened the pull request:
This PR uses a new `/block/:hash/txs` endpoint added in https://github.com/mempool/electrs/pull/24 to speed up block processing by loading block transactions in a single request instead of individually.

wiz reviewed (approved):
Tested ACK @ d35d04b2e82f134f90ae2875d368326a2b7a5585

softsimon reviewed (changes requested):
We must use the existing mempool transactions on new block

softsimon reviewed (changes requested):
The fee spans are now gone

<img width="652" alt="Screenshot 2023-07-25 at 00 26 02" src="https://github.com/mempool/mempool/assets/8561090/c736c13f-9acc-4206-91cf-657895503fc8">

mononaut commented:
Rebased on master, and refactored the `$getTransactionsExtended` function as follows:
 - try to look up transactions in the mempool cache first.
 - if esplora is enabled and we still need to fetch > 500 transactions, fetch them in bulk using the new mempool/electrs endpoint.
 - fetch any remaining missing transactions individually.
 
The other logic should be unchanged, except that I've removed the `memPool.hasPriority()` short-circuit on non-esplora backends. With the more conservative inSync behavior from #4043 the mempool always has priority while resyncing the last `INITIAL_BLOCKS_AMOUNT` blocks, so the short-circuit just wrecks those block templates.

softsimon reviewed (approved):
tested ACK @ [25925751e]
